x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -556,7 +591,15 @@ export class TaskRunner<
     } catch (err) {
       throw new ErrorWithReason(AlertExecutionStatusErrorReasons.License, err);
     }
+
+    if (rule.monitoring) {
+      if (rule.monitoring.execution.history.length >= MONITORING_HISTORY_LIMIT) {
```

Comment:
I am wondering if we should have `MONITORING_HISTORY_LIMIT` has a configuration, what do you think?

Reply:
I had that thought as well, but wasn't sure if users would care about configuring it. I can definitely add it if others feel strongly

Reply:
@kobelb what do you think?

Reply:
I think it's fine to use a constant for now. We can make it configurable in the future if we need to.